server/db/models/player.js

```diff
@@ -148,204 +89,183 @@ playerSchema.methods.setCardinal = function(row, col) {
+
+  if (currentTile[currentKey] === null && nextTile[nextKey] === null) return true;
+  else return false;
+}
+
+
+//do we need to add anything else to this function?
+//potentially use to deal with pushing adjacent players
+playerSchema.methods.oneCardMove = function(bearing) {
+  var newRow, newCol, opponent;
+
+  var checkMove = this.checkMove()
+
+  if (checkMove === true) {
+    var adjacentOpponent = this.game.getPlayerAt([this.position[0]+this.bearing[0], this.position[1]+this.bearing[1]])
+    if (adjacentOpponent) return adjacentOpponent.boardMove(this.bearing)
+    else return 
+  }
 }
 
+
 playerSchema.methods.cardMove = function (magnitude) {
   var newCol = this.position[1];
   var newRow = this.position[0];
 
-  // check that move is permitted
-  return this.checkMove()
-  .then(function(result){
-    if (result === true){
+  var checkMove = this.checkMove()
 
+  if (checkMove === true) {
+    while (magnitude > 0) {
+      newCol += this.bearing[1];
+      if (newCol < 0 || newCol > 11) return this.loseLife();
+
+      newRow += this.bearing[0];
+      if (newRow < 0 || newRow > 15) return this.loseLife();
+
+      var checkPit = this.checkForPit(newRow, newCol);
+      if (checkPit === true) return this.loseLife();
+
+      magnitude--
     }
-  })
-  .then(function(result) {
-    if (result === true) {
-      while(magnitude > 0){
-        newCol += this.bearing[1];
-        if (newCol < 0) return this.loseLife()
-        newRow += this.bearing[0];
-        if (newRow < 0) return this.loseLife()
-        magnitude --;
-      }
-      this.set('position', [newRow, newCol]);
-    }
-  })
-}
 
+    this.position = [newRow, newCol]
+
+    this.touchFlag(newRow, newCol)
+  }
+}
 
 
 playerSchema.methods.checkForPit = function(row, col) {
-  return this.findMyTile(row, col)
-  .then(function(tile) {
-    if (tile.floor === 'pit') return true;
-    else return false;
-  })
+  var tile = this.game.getTileAt(row, col)
+  if (tile.floor === 'pit') return true;
+  else return false;
 }
 
+
 playerSchema.methods.touchFlag = function(row, col) {
-  var player = this;
-  return player.findMyTile(row, col)
-  .then(function(tile) {
-    if (tile.flag != null) {
-      if (player.flagCount + 1 === tile.flag) {
-        player.flagCount++;
-        return player.save()
-      }
+  var tile = this.game.getTileAt(row, col)
+  if(tile.flag != null) {
+    if(this.flagCount + 1 === tile.flag) {
+      this.flagCount++;
+      this.checkIfWinner()
     }
-  })
-  .then(function(savedPlayer) {
-    return savedPlayer.checkIfWinner()
-  })
-  .then(function(result) {
-    if (result === true) return player.gameover()
-  })
+  }
 }
 
+
 playerSchema.methods.checkIfWinner = function() {
-  var player = this;
-  return mongoose.model('Game').findById(player.game)
-  .then(function(game) {
-    if (game.numFlags === player.flagCount) return true;
-    else return false;
-  })
+  if (this.game.numFlags === this.flagCount) this.gameover();
+  else return;
```

Comment:
are we keeping a reference to game on the player, or the other way around? if it's the other way around, we should probably check for wins in the game.